File: src/org/cappuccino_project/ide/intellij/plugin/psi/utils/ObjJNamedPsiUtil.java
package org.cappuccino_project.ide.intellij.plugin.psi.utils;

import com.intellij.psi.PsiElement;
import org.cappuccino_project.ide.intellij.plugin.psi.ObjJElementFactory;
import org.cappuccino_project.ide.intellij.plugin.psi.ObjJVariableName;
import org.jetbrains.annotations.NotNull;

public class ObjJNamedPsiUtil {

    public static boolean hasText(@NotNull ObjJVariableName variableNameElement, @NotNull String variableNameString) {
        return variableNameElement.getText().equals(variableNameString);
    }

    @NotNull
    public static String getName(@NotNull
                                         ObjJVariableName variableName) {
        return variableName.getText();
    }

    @NotNull
    public static PsiElement setName(@NotNull ObjJVariableName variableName, @NotNull String newName) {
        PsiElement newElement = ObjJElementFactory.createVariableName(variableName.getProject(), newName);
        variableName.replace(newElement);
        return newElement;
    }
}

File: src/cappuccino/ide/intellij/plugin/psi/utils/ObjJNamedPsiUtil.kt
package cappuccino.ide.intellij.plugin.psi.utils

import cappuccino.ide.intellij.plugin.psi.ObjJClassName
import com.intellij.psi.PsiElement
import cappuccino.ide.intellij.plugin.psi.ObjJElementFactory
import cappuccino.ide.intellij.plugin.psi.ObjJInstanceVariableDeclaration
import cappuccino.ide.intellij.plugin.psi.ObjJVariableName

object ObjJNamedPsiUtil {

    fun hasText(variableNameElement: ObjJVariableName, variableNameString: String): Boolean {
        return variableNameElement.text == variableNameString
    }

    fun getName(variableName: ObjJVariableName): String {
        return variableName.stub?.variableName ?: variableName.text
    }

    fun setName(variableName: ObjJVariableName, newName: String): PsiElement {
        val newElement = ObjJElementFactory.createVariableName(variableName.project, newName)
        return variableName.replace(newElement)
    }

    fun setName(instanceVariableDeclaration: ObjJInstanceVariableDeclaration, newName: String): PsiElement {
        val newElement = ObjJElementFactory.createVariableName(instanceVariableDeclaration.project, newName)
        return instanceVariableDeclaration.variableName?.replace(newElement)!!
    }

    fun getName(className: ObjJClassName): String {
        return className.stub?.getClassName() ?: className.text
    }

    fun setName(className: ObjJClassName, newClassName: String?): PsiElement {
        if (newClassName == null || newClassName.isEmpty()) {
            return className
        }
        val newElement: ObjJClassName = ObjJElementFactory.createClassName(className.project, newClassName) ?: return className
        return className.replace(newElement)
    }
}
